integrations/ollama/src/haystack_integrations/components/generators/ollama/chat/chat_generator.py

```diff
@@ -234,118 +270,168 @@ def from_dict(cls, data: Dict[str, Any]) -> "OllamaChatGenerator":
+
+                last_seen_calls.append(tc)
+
+        # ── merge text ────────────────────────────────────────────────
+        combined_text = "".join(c.content for c in chunks)
+
+        # ── rebuild completed ToolCall objects ────────────────────────
+        completed_tool_calls: List[ToolCall] = []
+        if last_seen_calls:
+            seen_ids = set()
+            # walk backwards so earlier duplicates are skipped
+            for tc in reversed(last_seen_calls):
+                tc_id = tc.get("id") or tc["function"]["name"]
+                if tc_id in seen_ids:
+                    continue
+                seen_ids.add(tc_id)
+
+                completed_tool_calls.append(
+                    ToolCall(
+                        tool_name=tc["function"]["name"],
+                        arguments=tool_call_args_by_id.get(tc_id),
+                    )
+                )
+            completed_tool_calls.reverse()  # restore original order
+
+        reply = ChatMessage.from_assistant(
+            text=combined_text,
+            tool_calls=completed_tool_calls or None,
+        )
 
         # Convert the last chunk's metadata to OpenAI format and attach it to the ChatMessage
         if chunks:
-            meta = _convert_ollama_meta_to_openai_format(chunks[-1].meta)
-            reply._meta = meta
+            reply._meta = _convert_ollama_meta_to_openai_format(chunks[-1].meta)
 
         return {"replies": [reply]}
 
+    # ------------------------------------------------------------------
+    # Main public entry point
+    # ------------------------------------------------------------------
     @component.output_types(replies=List[ChatMessage])
     def run(
-        self,
-        messages: List[ChatMessage],
-        generation_kwargs: Optional[Dict[str, Any]] = None,
-        tools: Optional[Union[List[Tool], Toolset]] = None,
-        *,
-        streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
+            self,
+            messages: List[ChatMessage],
+            generation_kwargs: Optional[Dict[str, Any]] = None,
+            tools: Optional[Union[List[Tool], Toolset]] = None,
+            *,
+            streaming_callback: Optional[Callable[[StreamingChunk], None]] = None,
     ):
         """
         Runs an Ollama Model on a given chat history.
 
         :param messages:
-            A list of ChatMessage instances representing the input messages.
+            Complete chat history, including the user’s latest message.
         :param generation_kwargs:
-            Optional arguments to pass to the Ollama generation endpoint, such as temperature,
-            top_p, etc. See the
-            [Ollama docs](https://github.com/jmorganca/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values).
+            Per-call overrides for Ollama inference options.  These are merged
+            on top of the instance-level `generation_kwargs`.
```

Comment:
I see the additional detail you'd like to add, but let's also keep the previous docstring that includes a link to the Ollama docs. So please combine the two together.